### PR TITLE
Initial checkin jms FAT to checkpoint/restore

### PR DESCRIPTION
**Pull request by sebratton**
Some existing FATS have been retrofitted to operate in checkpoint/restore mode.

### CODE REVIEW

**Comment by sebratton** [comment]
#build

File: dev/io.openliberty.checkpoint_fat_jms/bnd.bnd [hunk 1]
```diff
@@ -0,0 +1,53 @@
+#*******************************************************************************
+# Copyright (c) 2018, 2022 IBM Corporation and others.
```
Comment (dazavala) [inline dev/io.openliberty.checkpoint_fat_jms/bnd.bnd:2]: New part. Update copyright to 2023.

File: dev/io.openliberty.checkpoint_fat_jms/build.gradle [hunk 2]
```diff
@@ -0,0 +1,15 @@
+/*******************************************************************************
+ * Copyright (c) 2018,2020 IBM Corporation and others.
```
Comment (dazavala) [inline dev/io.openliberty.checkpoint_fat_jms/build.gradle:2]: Update copyright.

File: dev/io.openliberty.checkpoint_fat_jms/fat/src/io/openliberty/checkpoint/messaging/JMS20/fat/BucketSet1CpClientTest.java [hunk 3]
```diff
@@ -0,0 +1,287 @@
+/*******************************************************************************
+ * Copyright (c) 2016, 2020 IBM Corporation and others.
```
Comment (dazavala) [inline dev/io.openliberty.checkpoint_fat_jms/fat/src/io/openliberty/checkpoint/messaging/JMS20/fat/BucketSet1CpClientTest.java:2]: New part.  Review and update copyrights for all FAT sources.

File: dev/io.openliberty.checkpoint_fat_jms/bnd.bnd [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+# 
+# SPDX-License-Identifier: EPL-2.0
+#
+# Contributors:
+#     IBM Corporation - initial API and implementation
+#*******************************************************************************
+-include= ~../cnf/resources/bnd/bundle.props
+bVersion=1.0
+
+src:\
```
Comment (dazavala) [inline dev/io.openliberty.checkpoint_fat_jms/bnd.bnd:16]: FYI: All application src paths do not appear here.  Assuming that's on-purpose.  File `.classpath` looks complete in this regard.

File: dev/io.openliberty.checkpoint_fat_jms/bnd.bnd [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+#
+# FAT projects ("*_fat*") are automatically given dependencies on:
+#   junit
+#   fattest.simplicity
+#   componenttest-1.0
+
+-buildpath:\
+  com.ibm.websphere.javaee.annotation.1.1;version=latest,\
+  com.ibm.ws.logging;version=latest,\
+  com.ibm.websphere.javaee.servlet.3.0;version=latest,\
```
Comment (dazavala) [inline dev/io.openliberty.checkpoint_fat_jms/bnd.bnd:47]: We're starting at servlets 3.1 for EE8, yes? Also,  EJB 3.2, CDI 2.0, transaction 1.2 (1.1 is EE6, but compatible with EE7, EE8).

File: dev/io.openliberty.checkpoint_fat_jms/.classpath [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+	<classpathentry kind="src" path="test-applications/DurableUnshared/src"/>
+	<classpathentry kind="src" path="test-applications/JMSConsumer_118076/src"/>
+	<classpathentry kind="src" path="test-applications/JMSConsumer_118077/src"/>
+	<classpathentry kind="src" path="test-applications/JMSContextInject/src"/>
+	<classpathentry kind="src" path="test-applications/JMSDCF/src"/>
+	<classpathentry kind="src" path="test-applications/JMSDCFVar/src"/>
+	<classpathentry kind="src" path="test-applications/JMSMBeans/src"/>
+	<classpathentry kind="src" path="test-applications/JMSProducer/src"/>
+	<classpathentry kind="src" path="test-applications/JMSProducer_118073/src"/>
+	<classpathentry kind="src" path="test-applications/JMSRedelivery_120846/src"/>
```
Comment (dazavala) [inline dev/io.openliberty.checkpoint_fat_jms/.classpath:9]: I don't see any sources for JMSRedelivery_120846.  Also, after pulling your branch, I see the application dir exists but is empty when loading the FAT into eclipse.

If you are manually importing test app files from another FAT project, and you're using them as is, it's not too difficult to import them via build.gradle.  Doing this with MDB tests.

File: dev/io.openliberty.checkpoint_fat_jms/fat/src/io/openliberty/checkpoint/messaging/JMS20/fat/FATSuite.java [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.function.Consumer;
+
+import org.junit.ClassRule;
+import org.junit.runner.RunWith;
+import org.junit.runners.Suite;
+import org.junit.runners.Suite.SuiteClasses;
+
+import componenttest.annotation.SkipIfCheckpointNotSupported;
```
Comment (dazavala) [inline dev/io.openliberty.checkpoint_fat_jms/fat/src/io/openliberty/checkpoint/messaging/JMS20/fat/FATSuite.java]: Eclipse whines about many unused imports here. Eclipse reports some unused imports in other sources, fyi.

File: dev/io.openliberty.checkpoint_fat_jms/fat/src/io/openliberty/checkpoint/messaging/JMS20/fat/FATSuite.java [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+
+                  BucketSet1CpClientTest.class,
+                  BucketSet2CpClientTest.class,
+                  BucketSet1CpEngineTest.class,
+                  BucketSet2CpEngineTest.class,
+
+})
+
+public class FATSuite {
+
```
Comment (dazavala) [inline dev/io.openliberty.checkpoint_fat_jms/fat/src/io/openliberty/checkpoint/messaging/JMS20/fat/FATSuite.java:40]: As you go, clean up this class.  Editor shows lots of unnecessary whitespace, mixed use of tabs and spaces.  Methods look good.

File: dev/io.openliberty.checkpoint_fat_jms/publish/servers/LiteSet1Client/bootstrap.properties [hunk 9]
```diff
@@ -0,0 +1,3 @@
+bootstrap.include=../testports.properties
```
Comment (dazavala) [inline dev/io.openliberty.checkpoint_fat_jms/publish/servers/LiteSet1Client/bootstrap.properties:1]: FYI: Checkpoint trace and thread dump Xtrace are not enabled on these serves.

**Review by dazavala — DISMISSED** [review]
Aproved, @sebratton, as I see no issues that should prevent delivery of your initial FAT work.  I provided comments regarding copyright updates, .bnd bundle dependencies, issues discovered examining this branch in eclipse, and suggestions regarding server diagnostics and importing existing apps into this FAT project.  For instance, if you import test apps using gradle, we can ignore copyrights (i.e. dual maintenance).  Test changes related to InstantOn function look good.  I'll monitor this PR in case you push subsequent changes.

File: dev/io.openliberty.checkpoint_fat_jms/bnd.bnd (hunk 5 above)
Comment (dazavala) [reply dev/io.openliberty.checkpoint_fat_jms/bnd.bnd:47]: Take as an FYI to bump these versions from EE7 to EE8.

File: dev/io.openliberty.checkpoint_fat_jms/publish/servers/LiteSet1Client/bootstrap.properties (hunk 9 above)
Comment (dazavala) [reply dev/io.openliberty.checkpoint_fat_jms/publish/servers/LiteSet1Client/bootstrap.properties:1]: TBD

**Review by dazavala — DISMISSED** [review]
Conversations resolved, approved.

**Comment by sebratton** [comment]
#build

**Comment by sebratton** [comment]
#libby

**Comment by sebratton** [comment]
#build

**Comment by sebratton** [comment]
#libby

**Review by tjwatson — APPROVED** [review]

**Comment by sebratton** [comment]
#build
#libby
